File: deeplynx.models/ResponseDTOs/DataSourceResponseDto.cs
using System.Text.Json.Nodes;

namespace deeplynx.models;

public class DataSourceResponseDto
{
    public long Id { get; set; }
    public string Name { get; set; }
    public string? Description { get; set; }
    public bool Default { get; set; }
    public string? Abbreviation { get; set; }
    public string? Type { get; set; }
    public string? BaseUri { get; set; }
    public JsonObject? Config { get; set; }
    public long? ProjectId { get; set; }
    public DateTime LastUpdatedAt { get; set; }
    public long? LastUpdatedBy { get; set; }
    public bool IsArchived { get; set; }
}

File: deeplynx.UI/src/app/(home)/types/responseDTOs/dataSourceResponseDto.ts
export type DataSourceResponseDto = {
  id: number;
  name: string;
  description: string | null;
  default:boolean;
  abbreviation: string | null;
  type: string | null;
  baseuri: string | null;
  config: Record<string, unknown> | null; // object | null
  projectid: number;
  lastUpdatedAt: string | null; // RFC 3339 or null
  lastUpdatedBy: string | null;  
  isArchived:boolean;
  createdby: string | null;
  createdat: string;          // RFC 3339 date-time
  archivedat: string | null;  // RFC 3339 or null
};